com.unity.ml-agents/Runtime/Actuators/ActuatorManager.cs

```diff
@@ -0,0 +1,406 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Unity.MLAgents.Actuators
+{
+    /// <summary>
+    /// A class that manages the delegation of events, action buffers, and action mask for a list of IActuators.
+    /// </summary>
+    internal class ActuatorManager : IList<IActuator>
+    {
+        // IActuators managed by this object.
+        IList<IActuator> m_Actuators;
+
+        // An implementation of IDiscreteActionMask that allows for writing to it based on an offset.
+        IDiscreteActionMask m_DiscreteActionMask;
+
+        /// <summary>
+        /// Flag used to check if our IActuators are ready for execution.
+        /// </summary>
+        /// <seealso cref="ReadyActuatorsForExecution(IList{IActuator}, int, int, int)"/>
+        bool m_ReadyForExecution;
+
+        /// <summary>
+        /// The sum of all of the discrete branches for all of the <see cref="IActuator"/>s in this manager.
+        /// </summary>
+        internal int SumOfDiscreteBranchSizes { get; private set; }
```

Comment:
SumOfDiscreteBranchSizes, NumDiscreteBranches, NumContinuousActions is also part of ActionSpaceDef. Is it worth it to have an interface for these 3 properties shared between ActuatorManager and ActionSpaceDef?

Reply:
yeah, we could have `IActionSpecs`

Reply:
this was removed